torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -169,6 +172,15 @@ bool isSupported(Node* node) {
         return false;
       }
     }
+    if (node->isMemberOf(cuda_only_operator_set)) {
+      auto device = tensorexpr::pickDeviceType(node->inputs());
+      if (!device) {
+        device = tensorexpr::pickDeviceType(node->outputs());
```

Comment:
I think we require that the types be fully known for inputs & outputs, so in theory this conditional shouldn't matter.

Reply:
The reason I put it there is that maybe we could have an op with 0 tensor inputs - in this case we won't be able to deduce device from there. I don't think it could happen now, but better safe than sorry :)

Reply:
Oh, the other thing is that we're no longer checking that `exponent` has a known int / float type, which we need to know to compile it

Reply:
I'm not sure I follow: this change doesn't relax existing checks, only adds a new device check. So, if we don't check it now, we didn't check it before.

Also, this PR adds a test that iterates over all dtypes for the cases we do fuse - and it seems that on cuda the `pow.Tensor_Scalar` works correctly for all dtypes except float16 (which we should fix separately).

Reply:
yep, brain fryed from being the oncall 🤡  you're right we're still doing those checks